### PR TITLE
add datastore resets

### PR DESCRIPTION
**Pull request by coryschwartz**
This does not use the archive service locks, but will do datastore resets on a schedule.

### CODE REVIEW

**Comment by travisperson** [comment]
Why is this part of this helm chart?

File: charts/filecoin-chain-archiver/templates/reset-cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          containers:
+          - name: reset
+            image: bitnami/kubectl
+            command: ["bash", "-c"]
+            args:
+              - |
+                set -xe
+
+                kubectl exec {{ $reset.pod }} -- touch /var/lib/lotus/datastore/_reset
+                kubectl delete {{ $reset.pod }}
```
Comment (travisperson) [inline charts/filecoin-chain-archiver/templates/reset-cronjob.yaml:65]: I think we need to specify the namespace?

**Review by travisperson — COMMENTED** [review]
We might be able to deploy this without the node locker check if we pick the cron schedule times carefully to reset nodes on the same interval that snapshots are created. However, this still does run the risk of taking down a pod right at the end of a snapshot which could result in a page later due to a snapshot being too old.

**Comment by ognots** [comment]
[This dashboard](https://protocollabs.grafana.net/d/OftAaNVVz/chain-snapshot-metrics?orgId=1&from=now-24h&to=now), shows the average execution time is 2.2h (max 2.9h) over last 7 days, but that's only because we had an outage caused by leaving the datastore reset too late.

The active jobs from [this dashboard](https://protocollabs.grafana.net/d/7m4QV5R4z/chain-snapshots-overview?orgId=1) still show most job execution in under an hour, with some at around 1.5h

Given this data
* we can tolerate a job failure caused by this cronjob and still stay within SLA
* the risk of delaying landing threatens are SLA far more

My vote is to merge this, as is, and add the node locker functionality if needed at a later time.

**Review by ognots — APPROVED** [review]

**Comment by travisperson** [comment]
> The active jobs from [this dashboard](https://protocollabs.grafana.net/d/7m4QV5R4z/chain-snapshots-overview?orgId=1) still show most job execution in under an hour, with some at around 1.5h

I don't think this is correct. Which graph are you using to draw the conclusion that most job execution is under an hour? If you are looking at the `Active Jobs` dashboard it's showing that there is always at least one job running and for 30 minutes there are two, which would indicate that the execution is around 150 minutes or 2.5 hours

**Comment by ognots** [comment]
@travisperson good call, I was reading the graph wrong.
@coryschwartz please check the nodelocker. using that wait function you added is likely the best solution to keep logic out of the clean up job https://github.com/filecoin-project/filecoin-chain-archiver/pull/52

**Review by ognots — APPROVED** [review]

**Review by travisperson — COMMENTED** [review]
I'm worried about the dual infinite loops in this approach, and working around their issues may cause weird edge cases that are hard to handle.

As long as we can ensure that we do not reset a pod when it's actually locked, I think the infinite loops can be worked around and may aid in monitoring as we could detect the presence of a reset pod running for more than X hours enabling us to get notified of an issue an manually handle the failures.

File: charts/filecoin-chain-archiver/templates/reset-cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            env:
+              - name: FCA_NODELOCKER_OPERATOR_API
+                value: /dns/{{ include "filecoin-chain-archiver.nodelocker.fullname" $ }}/tcp/5101
+            command: ["bash", "-c"]
+            args:
+              - |
+                UNIQUE=$(printf "reset-%x-%x" $RANDOM $RANDOM)
+
+                echo acquiring lock for {{ $reset.pod }} with password $UNIQUE
+                filecoin-chain-archiver nodelocker operator lock --wait {{ $reset.pod }} $UNIQUE
```
[the remark this answers is not in the document]
Comment (coryschwartz) [reply charts/filecoin-chain-archiver/templates/reset-cronjob.yaml:72]: You're right....

This makes the coordination more complicated still. But you're right, that's what we have to do.

**Pull request merged**